### PR TITLE
Add search macro

### PR DESCRIPTION
**Pull request by ckeisato**
This PR adds a button style to support icons as buttons and adds a generic search macro.

### CODE REVIEW

File: src/forms/forms.macros.html [hunk 1]
```diff
@@ -38,9 +38,24 @@
   * @param {string} [class] - Class name for style or JS targeting
   * @param {string} [id] - ID for JS targeting
 #}
-{% macro input_button(input_name, input_placeholder=null, button_content='', button_style='primary', button_icon=null, required='true', class=null, id=null) %}
-<div class="input-group {% if button_icon %}input-group--button-icon{% else %}input-group--button{% endif %}{% if class %} {{ class }}{% endif %}" {% if id %} id="{{ id }}"{% endif %}>
-  <input class="input" name="{{ input_name }}" placeholder="{{ input_placeholder }}" required="{{ required }}">
+{% macro input_group(input_name, input_placeholder=null, button_content='', button_style='primary', button_icon=null, required='true', class=null, id=null) %}
+<div class="input-group {% if class %} {{ class }}{% endif %}" {% if id %} id="{{ id }}"{% endif %}>
+  <input class="input" name="{{ input_name }}" id="{{ input_name }}" {%if input_placeholder %} placeholder="{{ input_placeholder }}" {% endif %}required="{{ required }}">
```
Comment (josephco) [inline src/forms/forms.macros.html]: spacing

File: src/forms/forms.macros.html [hunk 2]
```diff
@@ -38,9 +38,24 @@
...
   {{ button(button_content, style=button_style, icon=button_icon, type="submit") }}
 </div>
 {% endmacro %}
+
+
+{#
+  * Renders a Search input with a search cta
+  * @param {string} input_name - Name attribute for the input element
```
Comment (josephco) [inline src/forms/forms.macros.html]: simplify params to name, placeholder

File: src/forms/forms.macros.html [hunk 3]
```diff
@@ -28,19 +28,34 @@
...
   * @param {boolean} [required=true] - Boolean for whether or not the input is required
   * @param {string} [class] - Class name for style or JS targeting
   * @param {string} [id] - ID for JS targeting
 #}
-{% macro input_button(input_name, input_placeholder=null, button_content='', button_style='primary', button_icon=null, required='true', class=null, id=null) %}
-<div class="input-group {% if button_icon %}input-group--button-icon{% else %}input-group--button{% endif %}{% if class %} {{ class }}{% endif %}" {% if id %} id="{{ id }}"{% endif %}>
-  <input class="input" name="{{ input_name }}" placeholder="{{ input_placeholder }}" required="{{ required }}">
+{% macro input_group(name, placeholder=null, button_content='', button_style='primary', button_icon=null, required='true', class=null, id=null) %}
+<div class="input-group {% if class %} {{ class }}{% endif %}" {% if id %} id="{{ id }}"{% endif %}>
+  <input class="input" name="{{ name }}" id="{{ name }}"{% if placeholder %} placeholder="{{ placeholder }}" {% endif %}required="{{ required }}">
```
Comment (josephco) [inline src/forms/forms.macros.html]: the endif after 'placeholder' needs to have a space after

File: src/forms/forms.macros.html [hunk 4]
```diff
@@ -28,19 +28,34 @@
...
+{#
+  * Renders a Search input with a search cta
+  * @param {string} name - Name attribute for the input element
+  * @param {string} [placeholder] - Placeholder text for the input
+  * @param {string} [class] - Class name for style or JS targeting
+  * @param {string} [id] - ID for JS targeting
+#}
+{% macro search_input(name, placeholder=null, class=null, id=null) %}
+<form class="input-group search-input{% if class %} {{ class }}{% endif %}" {% if id %} id="{{ id }}"{% endif %}>
+  <input class="input" name="{{ name }}" id="{{ name }}"{% if placeholder %} placeholder="{{ placeholder }}" {% endif %}required>
```
Comment (josephco) [inline src/forms/forms.macros.html]: same on this line

File: src/forms/forms.macros.html (hunk 3 above)
Comment (ckeisato) [inline src/forms/forms.macros.html]: doesn't it maintain the space after the `"`?

Comment (ckeisato) [inline src/forms/forms.macros.html]: nvm I see

**Comment by ckeisato** [comment]
verbal dtm from @josephco

**Pull request merged**